### PR TITLE
Login/sign-up screen

### LINKED ISSUE #9: FE: Demo User
- Create a "demo user" functionality that will be used later to allow recruiters and managers to easily test the app's features without having to sign up
- Add a button to allow for signing in as a demo user
- This is effectively a login button with a hardcoded username and password
- Show button both on signup and login pages
- In production you should manually create a user and some test data that would best show off your project. Do this in development right now to test this feature

### LINKED ISSUE #12: FE: Signup / Login Pages
- Edit the Login and Sign up pages to match your mock up
- Ignore Forget password (we don't need that right now)
- The page is currently responsive, but make sure it works with your mockups

### PR DESCRIPTION
**Pull request by felix-quinn**
### What this PR does:
- Updates login/sign-up page as per the mock-up.
- Adds demo login button as per #9.
- Removes username fields from APIs to reflect its absence.

### Screenshots / Videos:
image

### Any information needed to test this feature:
- cd client
- npm run start

### Any issues with the current functionality:
- UI doesn't match mock-up exactly yet (e.g. non-centered text).
- Demo login button copy-pastes login code instead of calling Login (assuming it is possible).

### CODE REVIEW

File: client/src/pages/Login/Login.tsx [hunk 1]
```diff
@@ -51,9 +53,11 @@ export default function Login(): JSX.Element {
               </Grid>
             </Grid>
             <LoginForm handleSubmit={handleSubmit} />
```
Comment (dproc96) [inline client/src/pages/Login/Login.tsx:59]: Can you vertically center this form to match the mocks?

File: client/src/pages/Login/Login.tsx [hunk 2]
```diff
@@ -51,9 +53,11 @@ export default function Login(): JSX.Element {
...
+            <AuthDemo />
           </Box>
-          <Box p={1} alignSelf="center" />
         </Box>
+        <Divider />
+        <AuthHeader linkTo="/signup" asideText="Don't have an account?" btnText="Create" />
```
Comment (dproc96) [inline client/src/pages/Login/Login.tsx]: This is appearing underneath the bottom of my screen, so I would have to scroll to find it. Can you bring it up a bit?

**Review by dproc96 — CHANGES_REQUESTED** [review]
Screen Shot 2021-06-04 at 9 04 04 AM
I noticed quite a few warnings in AuthDemo, can you address those? Additionally, can you include the background image from the mocks? It's in the Google Drive

**Comment by bonnieli** [comment]
Also has a merge conflict so please be sure to resolve this before another set of eyes for code review :)

**Review by bonnieli — APPROVED** [review]

**Review by ahmed-belhadj — CHANGES_REQUESTED** [review]
Can we break #9 into a separate PR for easier review and integration into `main`?

File: client/src/interface/User.ts [hunk 3]
```diff
@@ -1,6 +1,5 @@
 export interface User {
   email: string;
-  username: string;
```
Comment (ahmed-belhadj) [inline client/src/interface/User.ts:3]: Why are we removing username here? I think we need it.

**Review by ahmed-belhadj — CHANGES_REQUESTED** [review]

File: client/src/interface/User.ts (hunk 3 above)
Comment (felix-quinn) [reply client/src/interface/User.ts:3]: The mock-up does not have a username field.

Comment (ahmed-belhadj) [reply client/src/interface/User.ts:3]: Ah ok, I think we'll have to make this change on the backend as well then.

Comment (sina-jamshidi) [reply client/src/interface/User.ts:3]: Yes this is part of Felix's task. Felix please make sure all of the auth functionalities are still working in this PR

**Comment by sina-jamshidi** [comment]
This PR was approved, why is it changed to a draft?

**Comment by ahmed-belhadj** [comment]
> This PR was approved, why is it changed to a draft?

I wasn't sure if it was ready to merge into `main`

**Review by sina-jamshidi — APPROVED** [review]
pending no errors and merge conflict resolved

**Review by ahmed-belhadj — APPROVED** [review]

**Pull request merged**